src/App.vue

```diff
@@ -1,23 +1,124 @@
 <template>
-  <v-app>
-    <v-content>
-      <Home :task-path="taskPath" />
-    </v-content>
-  </v-app>
+  <form id="cfl-volunteer-tasks" class="cfl-volunteer-tasks">
+    <output v-if="tasksError" class="error">
+      {{ tasksError }}
+    </output>
+    <TaskList v-else :tasks="tasks" v-model="selectedTaskIds" />
+
+    <ContactMethod v-model="contactMethod" />
+
+    <div>
+      <output
+        v-if="message"
+        :class="{
+          success: formState === FormState.SUCCESS,
+          error: formState === FormState.ERROR
+        }"
+      >{{ message }}</output>
+      <button
+        v-if="formState !== FormState.SUCCESS"
+        :disabled="formState === FormState.SUBMITTING"
+        @click="submit"
+      >Submit</button>
+    </div>
+  </form>
 </template>
 
-<script>
-import Home from './components/Home'
-import config from '../public/config'
+<script lang="ts">
+import Vue from 'vue'
+import TaskList from './components/TaskList.vue'
+import ContactMethod from './components/ContactMethod.vue'
+import validEmail from '@/util/valid-email'
+
+enum FormState {
+  INITIAL,
+  SUBMITTING,
+  SUCCESS,
+  ERROR
+}
 
-export default {
+export default Vue.extend({
+  name: 'App',
   components: {
-    Home: Home
+    TaskList, ContactMethod
+  },
+  props: {
+    tasksHref: {
+      type: String,
+      required: true
+    }
+  },
+  beforeMount () {
+    this.loadTasks()
   },
   data () {
     return {
-      taskPath: config.taskPath
+      FormState,
+      formState: FormState.INITIAL,
+      message: '',
+      contactMethod: { email: '' },
+      selectedTaskIds: [],
+      tasks: [],
+      tasksError: ''
+    }
+  },
+  methods: {
+    reset () {
+      this.selectedTaskIds.splice(0)
+      this.contactMethod.email = ''
+    },
+    submit () {
+      this.message = ''
+      this.formState = FormState.SUBMITTING
+
+      if (!this.validate()) {
+        this.formState = FormState.ERROR
+      }
+
+      if (this.formState !== FormState.ERROR) {
+        setTimeout(() => {
+          if (Math.random() < 0.5) {
+            this.message = 'An error occurred while submitting the form.'
+            this.formState = FormState.ERROR
+          } else {
+            this.reset()
+            this.message = 'Thank you for volunteering! We\'ll contact you soon.'
+            this.formState = FormState.SUCCESS
+          }
+        }, 2000)
+      }
+    },
+    validate () {
+      if (!validEmail(this.contactMethod.email)) {
+        this.message = 'Please enter a valid email address.'
+        return false
+      }
+      return true
+    },
+    async loadTasks () {
+      try {
+        const response = await fetch(this.tasksHref)
+        const responseJson = await response.json()
+        if (!response.ok) {
+          throw new Error(
+            `Server returned an error: ${JSON.stringify(responseJson)}`
```

Comment:
This is fine for now, but eventually we should try to recover or inform the user that something has gone wrong

Reply:
The error is caught and logged and a user-friendly message is shown to the user. 😉